### ISSUE TITLE
1.0 Angles have changed

### ISSUE THREAD

**Issue by Alicecomma** [NoImageAvailable/CombatExtended#571]
While the bullet rotation is now correct, its shadow still remains wrongly rotated. Furthermore, other methods in ProjectileCE use shotRotation (which has changed its angle things in 1.0) so code like below should also be checked.

```
Rot4.FromAngleFlat(shotRotation)
```

**Comment by NoImageAvailable** [comment]
Apparent issue with fragmentation

https://i.imgur.com/Yk5DrVP.gifv

**Comment by Alicecomma** [comment]
```
Added injury to MechanicalCentipedeBodyFifthRing but it should be impossible to hit it. pawn=Mech_Centipede206621 dinfo=(def=Bomb_Secondary, amount= 1, instigator=Mech_Lancer206615, angle=21.2)
Verse.Log:Error(String, Boolean)
Verse.Hediff_Injury:PostAdd(Nullable1)
Verse.HediffSet:AddDirect(Hediff, Nullable1, DamageResult)
Verse.Pawn_HealthTracker:AddHediff(Hediff, BodyPartRecord, Nullable1, DamageResult)
Verse.DamageWorker_AddInjury:FinalizeAndAddInjury(Pawn, Hediff_Injury, DamageInfo, DamageResult)
Verse.DamageWorker_AddInjury:FinalizeAndAddInjury(Pawn, Single, DamageInfo, DamageResult)
Verse.DamageWorker_AddInjury:ApplySpecialEffectsToPart(Pawn, Single, DamageInfo, DamageResult)
Verse.DamageWorker_AddInjury:ApplyDamageToPart_Patch1(Object, DamageInfo, Pawn, DamageResult)
Verse.DamageWorker_AddInjury:ApplyToPawn(DamageInfo, Pawn)
Verse.DamageWorker_AddInjury:Apply(DamageInfo, Thing)
Verse.Thing:TakeDamage(DamageInfo)
CombatExtended.Harmony.Harmony_DamageWorker_AddInjury_ApplyDamageToPart:Postfix(DamageInfo, Pawn)
Verse.DamageWorker_AddInjury:ApplyDamageToPart_Patch1(Object, DamageInfo, Pawn, DamageResult)
Verse.DamageWorker_AddInjury:ApplyToPawn(DamageInfo, Pawn)
Verse.DamageWorker_AddInjury:Apply(DamageInfo, Thing)
Verse.Thing:TakeDamage(DamageInfo)
CombatExtended.BulletCE:Impact(Thing)
CombatExtended.ProjectileCE:TryCollideWith(Thing)
CombatExtended.ProjectileCE:CheckCellForCollision(IntVec3)
CombatExtended.ProjectileCE:CheckForCollisionBetween()
CombatExtended.ProjectileCE:Tick()
Verse.TickList:Tick()
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()`
```

**Comment by NoImageAvailable** [comment]
How is that error related to angles? Looks like a completely separate issue RE Centipede bodypart coverage

**Comment by Alicecomma** [comment]
Rot4.FromAngleFlat(shotRotation) is the code in the raycasting part of ProjectileCE and I thought the shotRotation basis has changed or something causing it to check for impact in the wrong direction.

The error I posted was reported tagging me whether it's the 1.0 angle problem so I posted it here.

I don't see how the fragments relate to this specific problem either, so it could be that none of these reported errors are related to the angle system.
